opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -85,6 +85,7 @@ public synchronized void reset() {
     /**
      * Noop method called on {@link Span#finish()}.
      */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // API method
```

Comment:
How this produces a warning message? E.g. here https://travis-ci.org/opentracing/opentracing-java/builds/276509101

Reply:
Could we rather move it to class level and keep the sources clean?

Reply:
The WeakerAccess on `Propagator TEXT_MAP` could be pushed up to the anonymous inner class level without additional exposure.

Moving the `@SuppressWarnings({"WeakerAccess", "unused"})` on `onSpanFinished` up to the class level would suppress those conditions from far more code than required and is generally looked down upon as a code smell.

With respect to your first question, it looks like the opentracing-java library is not cleaning its build dir: ```[INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ opentracing-mock ---
[INFO] Nothing to compile - all classes are up to date
```
IDEs run this sort of static analysis all the time though.  Fixing this stuff prevents having to wade through a forest of warnings.

Reply:
WRT the Propagator impl - I was originally interpreting the public static vars as API but looking more into it it looks like that concept wouldn't make as much sense.  Is there a reason they are not `private static final`?

Reply:
Ah, it's not worth it to me.  I just restored the warnings.

Reply:
It helps with _propagation_ tests where you want to be sure that span context is present in the carrier.